### DISCUSSION TITLE
Slow query, how to optimize?

### DISCUSSION THREAD

**Discussion by shamsch** [prisma/prisma#14514] [Q&A]
Hi, 

This is roughly my schema 

``` prisma
model A {
  id Int @id @default(autoincrement())
  x  B[] @relation("first")
  y  B[] @relation("second")
}

model B {
  id_x      Int
  x         A?     @relation(name: "first", fields: [id_x], references: [id])
  something String
  another   String
  thing     String
  id_y      Int
  y         A?     @relation(name: "second", fields: [id_y], references: [id])
  id        Int    @id @default(autoincrement())
}


```

My DB is running on AWS RDS Postgres. I have 3 million rows in model B and 400 rows in model A tables. When I make this following query with Prisma, it's taking kinda long (20s) to get the result back. Is there way to make it faster? 

``` JavaScript 
if (search) {
        searchQuery = {
            OR: [
                {
                    something: {
                        contains: search,
                        mode: "insensitive"
                    },
                },
                {
                    another: {
                        contains: search,
                        mode: "insensitive"
                    },
                },
                {
                    thing: {
                        contains: search,
                        mode: "insensitive"
                    },
                },
            ],
        } 
    }
const res= await prisma.B.findMany({
        take: limit, //10
        skip: offset, //1
        where: {
                 searchQuery
}
    });

```

**Comment by nurul3101** [comment]
Hi @shamsch 👋 

Could you try executing the query after enabling the [metrics](https://www.prisma.io/docs/concepts/components/prisma-client/metrics) preview feature, Prisma metrics give you a detailed insight into how Prisma Client interacts with your database. You can use this insight to help diagnose performance issues with your application.

One another approach I could think of is that you could configure [logging](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging#log-to-stdout) to see the underlying query which is being executed by Prisma, and then try to analyze the query execution plan of that query through [explain](https://www.postgresql.org/docs/current/sql-explain.html) command. This should give insights on where is the most time being taken.

**Reply by shamsch** [reply]
I get this back from analyzing: 
```SQL 
QUERY PLAN

Limit  (cost=12592.40..12592.81 rows=10 width=76) (actual time=2410.759..2410.763 rows=10 loops=1)
   ->  Index Scan Backward using "Journey_pkey" on "Journey"  (cost=0.43..125927.99 rows=3124074 width=76) (actual time=1.819..2396.829 rows=312398
loops=1)
 Planning Time: 4.699 ms
 Execution Time: 2410.867 ms
(4 rows)
```

Any ideas?

**Reply by pantharshit00** [reply]
Hello @shamsch 

Sorry for late follow up here. If this is still relevant, please try adding indexes for the fields that you are searching. Schema currently indicates there are no indexes in place.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
